app/controllers/search_controller.rb

```diff
@@ -4,8 +4,14 @@ class SearchController < ApplicationController
   def index
     mail = params[:search][:email]
```

Comment:
This actually causes a fatal error if no search parameters are set.

What about something like:
```ruby
def index
  permitted = params.require(:search).permit(:email)
  email = permitted[:email].strip
  raise ActionController::ParameterMissing unless permitted.present?

  @matching_users = User.where(email: email).where.not(id: current_user.id)
rescue ActionController::ParameterMissing
  respond_to do |format|
    format.html { redirect_to allies_path }
    format.json { head :no_content }
  end
end
```